packages/transformers/react-refresh-wrap/src/ReactRefreshWrapTransformer.js

```diff
@@ -94,45 +93,44 @@ export default new Transformer({
     return [asset];
   },
 
-  async generate({asset, options}) {
-    let code = await asset.getCode();
+  generate({asset, ast, options}) {
```

Comment:
~As I understand, the code in `Transformation.js#runTransformer` uses the last transformer with a generate function, so this is actually not needed?~

But that would require this transformer to be run after a transformer that does contain the parse/generate code for Babel.
I guess the only solution would be a shared `@parcel/ast-babel` as an abstraction...

Reply:
Yeah I think Babel parse+generate code is duplicated here and across the JS and Babel transformers. A shared library would make sense.